docs/source/download.rst

```diff
@@ -9,45 +9,67 @@ Thetis requires installation of `Firedrake
-take up to one hour depending on your system.
+   **Installing Firedrake can take up to an hour depending on the system.**
 
-The Thetis source will be installed in the ``src/thetis`` subdirectory
-of your Firedrake install.
-In order to use Firedrake and Thetis you need to activate the Firedrake
+After installation, in order to use Firedrake and install Thetis you need to activate the Firedrake
 virtualenv::
 
-    source <your-firedrake-install-dir>/firedrake/bin/activate
+   source <your-firedrake-venv-dir>/bin/activate
 
-Using this install method you should
-**not** add Thetis to your ``PYTHONPATH``. Instead, Thetis will
-automatically be available to import whenever your Firedrake
-virtualenv is active.
+.. warning::
 
-If you have already installed Firedrake
----------------------------------------
+   **Please check that Firedrake is working before trying to install Thetis.**
 
-You can install Thetis in your Firedrake installation by
-activating the Firedrake virtualenv and running::
 
-    firedrake-update --install thetis
+Installing Thetis
+------------------
 
+.. warning::
 
-If you are using a shared, pre-installed Firedrake (such as on some clusters)
------------------------------------------------------------------------------
+   ``pip install thetis`` will **not** install the desired Thetis package!
 
-Check out the `Thetis <http://github.com/thetisproject/thetis>`_
-repository from Github.
-You then need to add the Thetis repository to your ``PYTHONPATH`` in the
-Firedrake virtualenv. You can do this with ``pip``::
+You can install Thetis in your Firedrake installation by activating the Firedrake virtualenv and running:
+
+.. code-block:: none
+
+   pip install git+https://github.com/thetisproject/thetis.git
+
+Using this install method, you should **not** add Thetis to your
+``PYTHONPATH``. Instead, Thetis will automatically be available to import whenever your Firedrake virtualenv is active.
+
+.. _editable-install:
+
+Editable install
+=============================================================================
+
+If you want to install Thetis from a local checkout of the repository that you can directly edit, update (pull) from github, switch branches, etc., it is recommended to use an editable install using:
+
+.. code-block:: none
+
+   git clone https://github.com/thetisproject/thetis
+   pip install -e <path_to_thetis>
+
+An IDE such as PyCharm will not recognize Thetis when installed in this fashion for any project outside the cloned repository,
+as the source is not in site-packages. It will still run, and if you would like to enable full code navigation, you
+can add the Thetis cloned repository as a content root, then add the ``thetis`` sub-directory as a sources root.
+
+.. _alternative-installation-methods:
+
+Alternative installation methods
+=================================
+
+As well as being installable through ``pip``, Firedrake also provides Docker containers which typically contain the
```

Comment:
This isn't currently true: https://github.com/firedrakeproject/firedrake/blob/master/docker/Dockerfile.firedrake#L12

I will get that fixed.

Reply:
https://github.com/firedrakeproject/firedrake/pull/4225